tests/rptest/tests/datalake/datalake_services.py

```diff
@@ -52,6 +52,31 @@ def setUp(self):
                           self.redpanda.si_settings)
             svc.start()
             self.query_engines.append(svc)
+        rest_mode = not self.catalog_service.filesystem_wrapper_mode
+        if rest_mode:
+            # Apply REST end point configurations in Redpanda
+            # and restart brokers
+            self.redpanda.set_cluster_config(
+                {
+                    "iceberg_catalog_type": "rest",
+                    "iceberg_rest_catalog_endpoint":
+                    self.catalog_service.catalog_url,
+                    "iceberg_rest_catalog_user_id": "panda-user",
+                    "iceberg_rest_catalog_secret": "panda-secret"
+                },
+                expect_restart=True)
+            self.redpanda.logger.debug(
+                "Restarting Redpanda for Catalog REST server configuration to take effect."
+            )
+            brokers = self.redpanda.nodes
+            self.redpanda.restart_nodes(brokers)
```

Comment:
nit: seems like DatalakeServices is trying to do a bit too much -- this feels like it should be in dealt with when the RedpandaService is constructed. Perhaps we need something like SISettings and use EndToEndTest?

Reply:
ya e2e test is definitely a simpler pattern to avoid a restart but I think the complication is the bucket creation

- Catalog service depends on bucket to be created/available before it boots up
- Redpanda / query engines depends on catalog to be up (to have the URI)
*** bucket is created during the Redpanda initialization.

I think the right fix is to decouple bucket creation from RedpandaService, then have Catalog service start with the bucket information and then start the remaining services.. the decoupling didn't seem as straightforward at a first glance with all the cloud abstractions but perhaps it is as simple as creating an s3 client in DatalakeServices and set `bypass_bucket_creation=True` ?

Reply:
I took at stab at it, seems to have worked but I"m not sure if anything else would break (particularly in the bucket cleanup as it is now owned by datalake services). lmk what you think

Reply:
> Now that we are about to have multiple catalog types (Polaris and potentially Lakeouse) it would be great to have an ability to configure that as well

Can you explain a bit more please? We already parameterize tests with different catalog implementations (datalake_e2e_test), so I think the natural way to do this is to add more catalog types to IcebergCatalogMode? I didn't fully follow the part about inverting dependencies.

Reply:
I was thinking about passing a catalog factory in the test constructor, this way we do not have to put all the logic in the `DatalakeServices` but have it separated in the factory classes/methods. We can always think about that later